### PR TITLE
[SYCL][UR] Update UR loader version

### PR DESCRIPTION
**Pull request by bmyates**
Signed-off-by: Brandon Yates <brandon.yates@intel.com>

### CODE REVIEW

File: sycl/plugins/unified_runtime/CMakeLists.txt [hunk 1]
```diff
@@ -4,7 +4,7 @@ if (NOT DEFINED UNIFIED_RUNTIME_LIBRARY OR NOT DEFINED UNIFIED_RUNTIME_INCLUDE_D
   include(FetchContent)
 
   set(UNIFIED_RUNTIME_REPO "https://github.com/oneapi-src/unified-runtime.git")
-  set(UNIFIED_RUNTIME_TAG d6af758779db6eebdc419fd5e249302f566eb5de)
+  set(UNIFIED_RUNTIME_TAG 7d22f806c45872dd6326a123915fe3c75be25b0e)
```
Comment (jandres742) [inline sycl/plugins/unified_runtime/CMakeLists.txt:7]: I think we should start having the discipline of updating only stable tags from unified runtime. So in this case, the next weekly tag containing this change. That way, we have always at least a standard practice of what we want to update and we know the changes we are bringing from unified-runtime are stable.

However, I understand the urgency of this fix.

@smaslov-intel : what do you think?

File: sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_loader_interface.cpp [hunk 2]
```diff
@@ -86,7 +86,6 @@ UR_DLLEXPORT ur_result_t UR_APICALL urGetEnqueueProcAddrTable(
   pDdiTable->pfnMemUnmap = nullptr;
   pDdiTable->pfnUSMMemcpy = nullptr;
   pDdiTable->pfnUSMPrefetch = nullptr;
-  pDdiTable->pfnUSMMemAdvise = nullptr;
```
Comment (jandres742) [inline sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_loader_interface.cpp:89]: why was this removed? this is still on UR

https://github.com/oneapi-src/unified-runtime/blob/416b86a2c3de9e7104cf3ba5d11f064eff9c10fe/include/ur_api.h#L5219

File: sycl/plugins/unified_runtime/CMakeLists.txt (hunk 1 above)
Comment (bmyates) [reply sycl/plugins/unified_runtime/CMakeLists.txt:7]: Yes we should define validation+promotion process for accepting new UR versions. 

Currently I do not think there is any additional validation being done on UR weekly tags. So they are not necessarily more stable than other commits.

Comment (smaslov-intel) [reply sycl/plugins/unified_runtime/CMakeLists.txt:7]: I agree we need improved stability guarantees going forward, what can that UR standalone testing be?
Of course, updating SYCL to use newer UR will additionally run pre-commit SYCL testing.

**Review by smaslov-intel — APPROVED** [review]

File: sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_loader_interface.cpp (hunk 2 above)
Comment (bmyates) [reply sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_loader_interface.cpp:89]: It was renamed,  this variable is not in table anymore

Comment (jandres742) [reply sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_loader_interface.cpp:89]: @bmyates  could you add the new name, instead of removing it?

https://github.com/oneapi-src/unified-runtime/blob/416b86a2c3de9e7104cf3ba5d11f064eff9c10fe/include/ur_ddi.h#L1042

```cpp
    ur_pfnEnqueueUSMPrefetch_t pfnUSMPrefetch;
    ur_pfnEnqueueUSMAdvise_t pfnUSMAdvise;
    ur_pfnEnqueueUSMFill2D_t pfnUSMFill2D;
```

Comment (bmyates) [reply sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_loader_interface.cpp:89]: added

**Review by jandres742 — APPROVED** [review]

File: sycl/plugins/unified_runtime/CMakeLists.txt (hunk 1 above)
Comment (smaslov-intel) [inline sycl/plugins/unified_runtime/CMakeLists.txt:7]: Take a few more UR changes:
```suggestion
  set(UNIFIED_RUNTIME_TAG 7fa23fc8e6fbe1ac1de00b37d8f74620034bcb04)
```

File: sycl/plugins/unified_runtime/CMakeLists.txt [hunk 3]
```diff
@@ -4,7 +4,7 @@ if (NOT DEFINED UNIFIED_RUNTIME_LIBRARY OR NOT DEFINED UNIFIED_RUNTIME_INCLUDE_D
   include(FetchContent)
 
   set(UNIFIED_RUNTIME_REPO "https://github.com/oneapi-src/unified-runtime.git")
-  set(UNIFIED_RUNTIME_TAG d6af758779db6eebdc419fd5e249302f566eb5de)
+  set(UNIFIED_RUNTIME_TAG 7fa23fc8e6fbe1ac1de00b37d8f74620034bcb04)
```
Comment (smaslov-intel) [inline sycl/plugins/unified_runtime/CMakeLists.txt:7]:
```suggestion
  set(UNIFIED_RUNTIME_TAG 7d22f806c45872dd6326a123915fe3c75be25b0e)
```

**Comment by smaslov-intel** [comment]
@bader : it looks like this PR did not run any functional testing, did it?

**Comment by bader** [comment]
SYCL end-to-end testing is missing. Please, read [this discussion](https://github.com/intel/llvm/pull/8976#issuecomment-1500521548) to understand the root cause and suggested solutions.

**Comment by smaslov-intel** [comment]
> SYCL end-to-end testing is missing. Please, read [this discussion](https://github.com/intel/llvm/pull/8976#issuecomment-1500521548) to understand the root cause and suggested solutions.

I see. @bmyates : please re-base

**Comment by bmyates** [comment]
> Contributor

rebased onto head of sycl

**Review by cperkinsintel — APPROVED** [review]

**Comment by againull** [comment]
RHEL build fails in CI - http://icl-jenkins2.sc.intel.com:8080/blue/organizations/jenkins/SYCL_CI%2Fintel%2FBuild_PR_RHEL/detail/Build_PR_RHEL/5489/pipeline/152
That's why PR can't be merged yet.
@bmyates  Could you please take a look?

**Comment by bmyates** [comment]
> RHEL build fails in CI - http://icl-jenkins2.sc.intel.com:8080/blue/organizations/jenkins/SYCL_CI%2Fintel%2FBuild_PR_RHEL/detail/Build_PR_RHEL/5489/pipeline/152 That's why PR can't be merged yet. @bmyates Could you please take a look?

I think RHEL issue should be fixed now.  

pre-merge still failing due to unrelated #9008

**Comment by smaslov-intel** [comment]
rebased in https://github.com/intel/llvm/pull/9053
